gateway/src/apicast/oauth/oidc.lua

```diff
@@ -209,6 +224,17 @@ function _M:transform_credentials(credentials, cache_key)
   return { app_id = app_id }, ttl, payload
 end
 
+function _M:get_client_id(jwt_payload)
+  if self.tmpl and self.tmpl_type == "liquid" then
+      return self.tmpl:render(jwt_payload)
+  end
+
+  if self.tmpl and self.tmpl_type == "plain" then
+    return jwt_payload[self.tmpl:render()]
+  end
+
+  return jwt_payload.azp
```

Comment:
Maybe extracting `azp` to a `default_claim_with_app_id` constant or similar would help.
`jwt_payload[default_claim_with_app_id]`
What do you think?